### DISCUSSION TITLE
Dreamer nx boot loops

### DISCUSSION THREAD

**Discussion by Dofyx** [moonglow/FlashForge_Marlin#78] [Q&A] [answered]
hi,
after update my dreamer nx firmware with mks release, my dreamer boot loops.
FF Firmware writer no detect dreamer during boot loops
I see the bootloader at startup in the windows devices manager during the boot loops , but no drivers installed and drivers search fail
he is identified VID_0315&PID_000A, but no match in flashforge drivers inf :

DreamerDeviceID1 = "VID_0315&PID_0001"
DreamerDeviceID = "VID_2B71&PID_0001"
FinderDeviceID = "VID_2B71&PID_0002"
GuiderDeviceID = "VID_2B71&PID_0003"
Guider2DeviceID = "VID_2B71&PID_0004"
InventorDeviceID = "VID_2B71&PID_0005"
Inventor2DeviceID = "VID_2B71&PID_0006"
CreatorMaxDeviceID = "VID_2B71&PID_00E7"
FinderPlusDeviceID = "VID_2B71&PID_0007"
Guider2SDeviceID = "VID_2B71&PID_0009"
DreamerNxDeviceID = "VID_2B71&PID_000A"
CreatorPro2DeviceID = ""VID_2B71&PID_000E"
FinderClasOhlsonDeviceID = "VID_2B71&PID_00EE"
CreatorMax2DeviceID = ""VID_2B71&PID_0011"

I tested various old versions of flashprint but no found this ID
How to get this driver ?

I followed wiki tutorial for use FF Firmware writer 

Thanks for your help

**Comment by moonglow** [comment, marked as answer]
Hi!

Did you install external SPI flash module for MKS firmware as it was described in WIKI? MKS firmware will not works without it.

You need to install drivers manually by seleted it from FlashPrint drivers directory ( just install drivers by selecting NX model ) or use Linux to update or revert firmware. Windows writer tools requires drivers to be installed first.

**Reply by Dofyx** [reply]
actually I saw too late (too amazed by the screenshots )

Linux and ff_flash_firmware.py solved my problem and color ui boot well

Thank for help and good job for your free alternative firmware
